erpnext/manufacturing/doctype/bom_update_log/bom_updation_utils.py

```diff
@@ -0,0 +1,224 @@
+# Copyright (c) 2022, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import json
+from collections import defaultdict
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+if TYPE_CHECKING:
+	from erpnext.manufacturing.doctype.bom_update_log.bom_update_log import BOMUpdateLog
+
+import frappe
+from frappe import _
+
+
+def replace_bom(boms: Dict) -> None:
+	"Replace current BOM with new BOM in parent BOMs."
+
+	current_bom = boms.get("current_bom")
+	new_bom = boms.get("new_bom")
+
+	unit_cost = get_bom_unit_cost(new_bom)
+	update_new_bom_in_bom_items(unit_cost, current_bom, new_bom)
+
+	frappe.cache().delete_key("bom_children")
+	parent_boms = get_ancestor_boms(new_bom)
+
+	for bom in parent_boms:
+		bom_obj = frappe.get_doc("BOM", bom)
+		# this is only used for versioning and we do not want
+		# to make separate db calls by using load_doc_before_save
+		# which proves to be expensive while doing bulk replace
+		bom_obj._doc_before_save = bom_obj
+		bom_obj.update_exploded_items()
+		bom_obj.calculate_cost()
+		bom_obj.update_parent_cost()
+		bom_obj.db_update()
+		if bom_obj.meta.get("track_changes") and not bom_obj.flags.ignore_version:
+			bom_obj.save_version()
```

Comment:
```suggestion
		bom_obj.save_version()
```

both condition already seem to be part of `Document.save_version`


Also check if version is getting created, since `_doc_before_save` is required which isn't triggered unless post save methods are run... maybe it can be added manually. 


PS: tested locally, no version docs found for bom.

Reply:
On second thought generating versions for all BOMs, every day seems a bit too much. 😅

Reply:
fixed locally, version was not created because the faux `_doc_before_save` pointed to the bom_obj so there was never any difference between the two. Just had to make a copy and assign to `_doc_before_save`